pkg/rke2/machine_filters.go

```diff
@@ -21,7 +21,7 @@ func matchesRCPConfiguration(infraConfigs map[string]*unstructured.Unstructured,
 // matchesRKE2BootstrapConfig checks if machine's RKE2ConfigSpec is equivalent with RCP's RKE2ConfigSpec.
 func matchesRKE2BootstrapConfig(machineConfigs map[string]*bootstrapv1.RKE2Config, rcp *controlplanev1.RKE2ControlPlane) collections.Func {
 	return func(machine *clusterv1.Machine) bool {
-		return true // TODO: implement
+		return true // TODO: implement by using inspiration from https://github.com/kubernetes-sigs/cluster-api/blob/v1.3.2/controlplane/kubeadm/internal/filters.go#L81
```

Comment:
Do we have an issue created for this todo? Does this need to be done before releasing?

Reply:
I would say:
- No, we don't have an issue yet, I will create it now.
- About releasing, that's a more challenging question. It is only necessary for upgrading theoretically. The question is then, should this release (or more exactly, pre-release) support upgrades or not. If yes, we need to do more testing, delaying the release itself.

Reply:
Let's take this bit as it is and we will resolve the outstanding issues with a further PR coming soon.